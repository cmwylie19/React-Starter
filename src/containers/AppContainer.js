import React, { Fragment } from 'react'
import { BrowserRouter, Link, Route } from 'react-router-dom';
import { ThemeProvider, AuthProvider } from '../reducers'
import { TopPanel, BottomPanel } from '../components/index'
import { RouterProvider } from '../reducers/RouterReducer'
import { UserForm } from '../components/UserForm'

import { Home } from '../components/Home'

export default function AppContainer() {
    return (
        <RouterProvider>
            <ThemeProvider>
            <AuthProvider>
                <TopPanel />
                <Fragment>
                    <Route component={UserForm} path="/" exact />
                    <Route component={Home} path="/home/:name" exact />
                </Fragment>
                <BottomPanel />
            </AuthProvider>
            </ThemeProvider>
        </RouterProvider>
    );
}