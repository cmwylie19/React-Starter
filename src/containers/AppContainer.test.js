import React from 'react';
import ReactDOM from 'react-dom';
import AppContainer from './AppContainer';
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AppContainer />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const tree = renderer
      .create(<AppContainer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });