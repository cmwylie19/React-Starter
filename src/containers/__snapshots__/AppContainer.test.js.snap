// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "color": "black",
        "display": "flex",
        "height": "70px",
        "justifyContent": "space-between",
        "left": 0,
        "padding": "15px",
        "position": "absolute",
        "right": 0,
        "width": "100% !important",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <img
        src="hatLogo.png"
        style={
          Object {
            "height": "80px",
          }
        }
      />
      <h2
        style={
          Object {
            "fontFamily": "Red Hat Display",
            "padding": "10px",
          }
        }
      >
        Big Case
      </h2>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <div>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "purple",
              "borderRadius": "15px",
              "fontSize": "1em",
              "height": "50px",
              "margin": "20px",
              "width": "120px",
            }
          }
        >
          Custom
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "orange",
              "borderRadius": "15px",
              "fontSize": "1em",
              "height": "50px",
              "margin": "20px",
              "width": "120px",
            }
          }
        >
          Orange
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderRadius": "15px",
              "fontSize": "1em",
              "height": "50px",
              "margin": "20px",
              "width": "120px",
            }
          }
        >
          Blue
        </button>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <img
          src="user.png"
          style={
            Object {
              "borderRadius": "20px",
              "height": "60px",
              "width": "60px",
            }
          }
        />
        <div>
          Example User
        </div>
      </div>
    </div>
  </div>,
  <form
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "height": "100vh",
        "justifyContent": "center",
        "textAlign": "left",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid black",
          "borderRadius": "15px",
          "padding": "10px",
        }
      }
    >
      <label>
        <h5
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          User Form
        </h5>
        Name: 
         
        <input
          name="name"
          onChange={[Function]}
          type="text"
          value="Example User"
        />
      </label>
      <br />
      <label>
         Avatar 
         
        <input
          onChange={[Function]}
          type="file"
        />
      </label>
      <br />
      <br />
      <br />
      <input
        onClick={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        type="submit"
        value="Submit"
      />
    </div>
  </form>,
  <div
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "purple",
        "bottom": 0,
        "color": undefined,
        "display": "flex",
        "height": "70px",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "width": "100% !important",
      }
    }
  >
    Send me an e-
    <a
      href="mailto: cwylie@redhat.com?&subject=SiteTemplate&body=Nice..."
    >
      <div>
         
      </div>
       mail
    </a>
  </div>,
]
`;
